### PR TITLE
Refs [TASK][34052] Handle event News Screen

### PR DESCRIPTION
**Pull request by trangnq258**
## Related Tickets
- [#Task 34052: Handle event News Screen](https://edu-redmine.sun-asterisk.vn/issues/34052)
## WHAT
- Bind Data : get data pour into recyclerview
- Item Click : show webview with clicking position item
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -13,20 +13,18 @@
...
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
-        <activity android:name=".ui.main.MainActivity">
-
-        </activity>
+        <activity android:name=".ui.main.MainActivity"></activity>
```
Comment (tuannm-1276) [inline app/src/main/AndroidManifest.xml]:
```suggestion
        <activity android:name=".ui.main.MainActivity"/>
```

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -13,20 +13,18 @@
         android:roundIcon="@drawable/splash"
         android:supportsRtl="true"
         android:theme="@style/Theme.Yummy">
+        <activity android:name=".ui.news.NewsActivity"></activity>
```
Comment (tuannm-1276) [inline app/src/main/AndroidManifest.xml]:
```suggestion
        <activity android:name=".ui.news.NewsActivity"/>
```

File: app/src/main/java/com/example/yummy/ui/favorite/AllMealFragment.kt [hunk 3]
```diff
@@ -3,10 +3,13 @@ package com.example.yummy.ui.favorite
 import com.example.yummy.R
 import com.example.yummy.base.BaseFragment
 
-class AllMealFragment :BaseFragment() {
+class AllMealFragment : BaseFragment() {
     override val layoutResource: Int
         get() = R.layout.fragment_favorite_all
 
+    override fun setupView() {
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/yummy/ui/favorite/AllMealFragment.kt]:
```suggestion
    override fun setupViews() {
```

File: app/src/main/java/com/example/yummy/ui/adapter/NewsAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.yummy.R
+import com.example.yummy.data.model.News
+
+class NewsAdapter : RecyclerView.Adapter<NewsViewHolder>() {
+
+    var onItemClick: (News) -> Unit = { _ -> }
+    private var news = mutableListOf<News>()
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/adapter/NewsAdapter.kt]:
```suggestion
    private val news = mutableListOf<News>()
```

File: app/src/main/java/com/example/yummy/ui/adapter/NewsAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    override fun getItemCount() = news.size
+
+    override fun onBindViewHolder(holder: NewsViewHolder, position: Int) {
+        holder.bindData(news[position])
+    }
+
+    fun updateData(newList: List<News>) {
+        news.let {
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/adapter/NewsAdapter.kt]:
```suggestion
        news.run {
```
scopefunction

File: app/src/main/java/com/example/yummy/ui/dialog/LoadingDialog.kt [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+class LoadingDialog(context: Context) : Dialog(context) {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        setContentView(R.layout.dialog_loading)
+        val window = this.window
+        window?.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
+        window?.setGravity(Gravity.CENTER)
+        setCancelable(false)
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/dialog/LoadingDialog.kt]: Dùng scope function

File: app/src/main/java/com/example/yummy/ui/news/NewsActivity.kt [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import android.content.Intent
+import android.webkit.WebViewClient
+import com.example.yummy.R
+import com.example.yummy.base.BaseActivity
+import com.example.yummy.utlis.BundleConst.LINK
+import kotlinx.android.synthetic.main.activity_news.*
+
+class NewsActivity : BaseActivity() {
+    override val layoutResource: Int
+        get() = R.layout.activity_news
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/news/NewsActivity.kt]:
```suggestion
    override val layoutResource get() = R.layout.activity_news
```

File: app/src/main/java/com/example/yummy/ui/news/NewsActivity.kt [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+    override fun initComponents() {
+        val url = intent?.getStringExtra(LINK)
+        webViewNews.webViewClient = WebViewClient()
+        url?.let { webViewNews.loadUrl(it) }
+    }
+
+    companion object {
+        fun getIntent(context: Context, link: String) =
+            Intent(context, NewsActivity::class.java).putExtra(LINK, link)
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/news/NewsActivity.kt]: xem lại convention về đặt tên EXTRA nhé
em đặt private const trong Activity này luôn vì nó chỉ dùng ở đây, không nhất thiết phải public 
https://github.com/framgia/coding-standards/tree/master/eng/android

Comment (huytq-2422) [reply app/src/main/java/com/example/yummy/ui/news/NewsActivity.kt]: 2.10

**Review by huytq-2422 — APPROVED** [review]

**Pull request merged**